modelon/impact/client/operations.py

```diff
@@ -201,17 +194,100 @@ def cancel(self):
         self._model_exe_sal.compile_cancel(self._workspace_id, self._fmu_id)
 
 
-class ExperimentOperation(Operation):
+class CachedModelExecutableOperation(Operation):
```

Comment:
I think we also should overwrite the `wait` method to make things go fast, I am thinking something like:
            
```
if self.status() != status:
     raise exceptions.OperationTimeOutError(f"The operation '{self.name}' have the status '{self.status().name}', it will never get the status '{status.name}'")

logger.info("Found cached FMU which will be used")
return self.data()
```

Reply:
Good Idea! But I was thinking of just adding a special case to the base class **wait()** function instead of overwriting it. If we overwrite it in the child class, we would have to keep a dummy 'timeout' parameter,which seemed a bit ugly to me?
I was thinking of something like this :-
```
        if self.name == "Looking for cached FMU":
            if self.status() != status:
                raise exceptions.OperationTimeOutError(
                    f"The operation '{self.name}' has the status '{self.status().name}', "
                    f"it will never get the status '{status.name}'!"
                )

            logger.info("Cached FMU found! Using the cached FMU!")
            return self.data()
```

Reply:
A base class should never need to know about implementation details in classes that inherents from it. That the timeout parameter would have no effect for the overriden method seems ok to me.